mariamihaelafainarea/assignment2/LowestAncestor

```diff
@@ -0,0 +1,102 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author maria
+ */
+public class LowestAncestor<T> {
+    
+    Node first;
+    //these parameters help me find
+    //if the two values searched were found
+    private boolean param1 = false;
```

Comment:
Try to get rid of param1 and param2. Here is the comment I left 3 days ago:

So basically you use param1 and param2 to make sure that both nodes exist in the tree, right? You can for example just check if both nodes exist in the tree before returning the value. Or alternatively keep a local variable in the method. Global variables should be avoided if they can. If they store some important state then they make sense, but in this case they are just an assertion of a single run of a method so they should be stored only locally.

If that is still not clear enough let me know and I will try to help in a more concrete way :)

Reply:
Okei i will try remove it or put it in a function. But if i try to search a key in the tree it will take more time and will increase complexity (The tree is not a binary tree)

Reply:
Okay that's already better :) But maybe try to pass two booleans instead of an array of Booleans? The reason for it is that with an array it can be easily misused: someone can pass a longer or shorter array and you still will not have the params that you expect. It is always supposed to be two boolean values, in every run of the findLowestAncestor method, so just add them as the expected parameters.

I agree with you that looking for the values beforehand will be slower. But it would still be better than introducing a possible bug :)